kubetest/azure.go

```diff
@@ -545,6 +565,27 @@ func (c *Cluster) populateApiModelTemplate() error {
 		}
 	}
 
+	// Populate PrivateAzureRegistryServer field if we are using ACR and custom-built k8s components
+	if strings.Contains(imageRegistry, "azurecr") && (*aksCustomK8sComponents || *aksHyperKube) {
+		v.Properties.OrchestratorProfile.KubernetesConfig.PrivateAzureRegistryServer = imageRegistry
+	}
+
+	if *aksCustomK8sComponents { // k8s 1.17+
```

Comment:
how do you know this is 1.17+?

Reply:
Based on https://github.com/Azure/aks-engine/pull/2191, 1.17 won't use hyperkube for kube-* components anymore. In the future, 1.17+ jobs should use `--aksengine-custom-k8s-components=True` flag instead of `--aksengine-hyperkube=True`.

Reply:
then, could we make a check of kubernetes version and decide the options based on the result?

Reply:
Updated the PR. PTAL